### PR TITLE
add options to tracer

### PR DESCRIPTION
**Pull request by tc-wilson**

### CODE REVIEW

File: tracer-client [hunk 1]
```diff
@@ -23,7 +24,7 @@ fi
 duration=60
 tracer=osnoise
 
-longopts="duration:,tracer:"
+longopts="duration:,tracer:,period:,runtime:,us_single:,us_total:,threshold:,sample:,noise:,cpu:,events:"
```
Comment (k-rister) [inline tracer-client:27]: `sample:` and `noise:` are defined here, but not below...

File: tracer-client [hunk 2]
```diff
@@ -38,6 +39,46 @@ while true; do
...
+	--threshold)
+	    shift
+	    threshold=$1
+	    shift
+	    ;;
+	--stack)
+	    shift
+	    stack=$1
+	    shift
+	    ;;
```
Comment (k-rister) [inline tracer-client:71]: ...and `stack` is defined here, but not above.

File: tracer-client [hunk 3]
```diff
@@ -78,10 +119,69 @@ esac
...
+           then
+              previous_us_total=`cat osnoise/stop_tracing_total_us`
+              echo "Setting osnoise/stop_tracing_total_us to '$us_total'"
+              echo $us_total > osnoise/stop_tracing_total_us
+        elif [ ${us_total} ] && [ -d osnoise/stop_tracing_out_us ]
+           then
+              previous_us_total=`cat osnoise/stop_tracing_out_us`
+              echo "Setting osnoise/stop_tracing_out_us to '$us_total'"
+              echo $us_total > osnoise/stop_tracing_out_us
+        fi
```
Comment (k-rister) [inline tracer-client:176]: more funky indentation

File: tracer-client [hunk 4]
```diff
@@ -101,14 +201,66 @@ esac
 	echo
 
 	echo "TRACE:"
-	cat trace
+        if [ -z ${cpu} ]; then cat trace; else cat per_cpu/cpu${cpu}/trace; fi
```
Comment (k-rister) [inline tracer-client:204]: expand into a block and fix indentation

File: tracer-client [hunk 5]
```diff
@@ -101,14 +201,66 @@ esac
...
+        elif [ ${previous_us_total} ] && [ -d osnoise/stop_tracing_out_us ]
+           then
+              echo "Restoring osnoise/stop_tracing_out_us to '$previous_us_total'"
+              echo $previous_us_total > osnoise/stop_tracing_out_us
+        fi
+
+        if [ ${previous_runtime} ]; then echo "Restoring osnoise/runtime_us to '$previous_runtime'"; echo "$previous_runtime" > osnoise/runtime_us; fi
+        if [ ${previous_threshold} ]; then echo "Restoring osnoise/tracing_threshold to '$previous_threshold'"; echo "$previous_threshold" > osnoise/tracing_threshold; fi
+        if [ ${previous_stack} ]; then echo "Restoring osnoise/print_stack to '$previous_stack'"; echo "$previous_stack" > osnoise/print_stack; fi
+
```
Comment (k-rister) [inline tracer-client:260]: more indentation and block expansion needed

**Comment by tc-wilson** [comment]
updated

**Comment by k-rister** [comment]
@tc-wilson 

So it looks like the indentation is still messed up.  I'm thinking this is an editor issue which we have solved before.  Try adding these two lines under the interpreter:

```
# -*- mode: sh; indent-tabs-mode: nil; sh-basic-offset: 4 -*-
# vim: autoindent tabstop=4 shiftwidth=4 expandtab softtabstop=4 filetype=bash
```

Then save and exit the file and reopen it.  Your editor should pickup the proper mode to use for editing the file and then you can fix your indentation.  I mostly know how this behaves from the emacs side, you may have to get some input from @atheurer on the vi side of things.  For example, for emacs I usually have to forceably update a whitespace formatting by modifying the line if I'm trying to fix it (emacs does not automatically fix all lines in the file).

See https://github.com/k-rister/crucible/blob/master/bin/_main for an example.

**Pull request merged**